controllers/vspheredeploymentzone_controller_domain.go

```diff
@@ -100,6 +100,16 @@ func (r vsphereDeploymentZoneReconciler) reconcileTopology(ctx context.Context,
 		}
 	}
 
+	for _, networkConfig := range topology.NetworkConfigurations {
+		if networkConfig.NetworkName == "" {
+			continue
+		}
+		if _, err := deploymentZoneCtx.AuthSession.Finder.Network(ctx, networkConfig.NetworkName); err != nil {
+			conditions.MarkFalse(deploymentZoneCtx.VSphereDeploymentZone, infrav1.VSphereFailureDomainValidatedCondition, infrav1.NetworkNotFoundReason, clusterv1.ConditionSeverityError, "network %s is misconfigured", networkConfig.NetworkName)
```

Comment:
is the network not found rather than misconfigured in such a scenario?

Reply:
Looks like it's copied from the original network config without failure domain. Could you update both places?

Reply:
Updated both